src/lib/Scenes/Artwork/Components/PartnerCard.tsx

```diff
@@ -0,0 +1,147 @@
+import { EntityHeader, Flex } from "@artsy/palette"
+import { PartnerCard_artwork } from "__generated__/PartnerCard_artwork.graphql"
+import { PartnerCardMutation } from "__generated__/PartnerCardMutation.graphql"
+import InvertedButton from "lib/Components/Buttons/InvertedButton"
+import SwitchBoard from "lib/NativeModules/SwitchBoard"
+import { filterLocations } from "lib/utils/filterLocations"
+import { get } from "lib/utils/get"
+import { limitWithCount } from "lib/utils/limitWithCount"
+import React from "react"
+import { TouchableWithoutFeedback } from "react-native"
+import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
+
+interface Props {
+  artwork: PartnerCard_artwork
+  relay: RelayProp
+}
+
+interface State {
+  isFollowedChanging: boolean
+}
+
+export class PartnerCard extends React.Component<Props, State> {
+  state = { isFollowedChanging: false }
+  handleFollowPartner = () => {
+    const { artwork, relay } = this.props
+    const {
+      gravityID: partnerSlug,
+      profile: { is_followed: partnerFollowed, internalID: profileID },
+    } = artwork.partner
+
+    this.setState(
+      {
+        isFollowedChanging: true,
+      },
+      () => {
```

Comment:
Nit: no need to wait for React to re-render the UI until you send off the network request, these things can be done in parallel.